instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/bedrock.py

```diff
@@ -472,6 +499,65 @@ def _on_stream_error_callback(
                 attributes=metrics_attributes,
             )
 
+    def _invoke_agent_on_success(
+        self,
+        span: Span,
+        result: dict,
+        instrumentor_context: _BotocoreInstrumentorContext,
+    ):
+        try:
+            if "completion" in result and isinstance(result["completion"], EventStream):
+                event_stream = result["completion"]
+
+                # Drain the stream so we can instrument AND keep events
+                all_events = list(event_stream)
+
+                # A replay generator so user code can still iterate
+                result["completion"] = _replay_events(all_events)
```

Comment:
if we use a wrapper that's an iterator, as mentioned above, I'm assuming we wouldn't need this line